File: Solution/WebSolution/Java/AssessmentAPI/src/main/java/com/transflower/tflassessment/services/CandidateAnswerService.java

package com.transflower.tflassessment.services;

import java.util.List;
import java.util.concurrent.CompletableFuture;

import com.transflower.tflassessment.entities.CandidateAnswer;
import com.transflower.tflassessment.entities.CandidateTestDetails;

public interface CandidateAnswerService {
    
    
    CompletableFuture<Boolean> insertCandidateAnswer(int candidateId, List<CandidateAnswer> answers);

    CompletableFuture<List<CandidateAnswer>> getCandidateAnswer(int candidateId, int testId);

    
    CompletableFuture<List<CandidateAnswer>> getCandidateAnswerResult(int candidateId, int testId);

    
    CompletableFuture<CandidateTestDetails> getCandidateTestDetails(int candidateId, int testId); 
}

File: Solution/WebSolution/Dotnet/AssessmentPortal/Microservices/CandidateAnswerAPI/Services/CandidateAnswerService.cs
using Transflower.TFLAssessment.Entities;
using Transflower.TFLAssessment.Services.Interfaces;
using Transflower.TFLAssessment.Repositories.Interfaces;

namespace Transflower.TFLAssessment.Services
{
    public class CandidateAnswerService : ICandidateAnswerService
    {
        private readonly ICandidateAnswerRepository _repository;

        public CandidateAnswerService(ICandidateAnswerRepository repository)
        {
            _repository = repository;
        }

        public Task<bool> InsertCandidateAnswers(int candidateId, List<CandidateAnswer> answers)
        {
            return _repository.InsertCandidateAnswers(candidateId, answers);
        }
        public Task<List<CandidateAnswer>> GetCandidateAnswers(int candidateId, int testId)
        {
            return _repository.GetCandidateAnswers(candidateId, testId);
        }
        public Task<List<CandidateAnswerResult>> GetCandidateAnswerResultsAsync(int candidateId, int testId)
        {
            return _repository.GetCandidateAnswerResultsAsync(candidateId, testId);
        }

        public Task<CandidateTestDetails> GetCandidateTestDetails(int candidateId, int testId)
        {
            return _repository.GetCandidateTestDetails(candidateId, testId);
        }
    }
}
